reference/components/cf-net.markdown

```diff
@@ -0,0 +1,138 @@
+---
+layout: default
+title: cf-net
+published: true
+sorting: 90
+tags: [Components, cf-net]
+keywords: [protocol, cli]
+---
+
+`cf-net` can be used to send simple protocol commands to a policy server.
+It is a Command-Line-Interface (CLI) to the CFEngine network protocol, and a standalone tool.
+`cf-net` is not needed or used by any of the other binaries.
+The tool can be used to send commands like `GET` and `OPENDIR` without writing policy.
+It is in some ways an extremely light-weight version of `cf-agent` - policy evaluation is replaced with easy to use command line arguments.
+
+## Command reference ##
+
+[%CFEngine_include_snippet(cf-net.help, [\s]*--[a-z], ^$)%]
+
+<!-- ** <- Terminate syntax highlighting of weird regex -->
+
+## Bootstrapping and cf-key
+
+`cf-net` *needs* a key-pair generated by `cf-key` to communicate with a server.
+Thus, the easiest way to use cf-net is on a successfully bootstrapped client:
+```
+$ sudo /var/cfengine/bin/cf-key
+$ sudo /var/cfengine/bin/cf-agent --bootstrap myhostname
+$ sudo /var/cfengine/bin/cf-net connect
+Connected & authenticated successfully to 'myhostname'
+```
+(`myhostname` can also be an IP address)
+
+All three commands above are run with sudo, so they access the same key file.
+In the future, we would like to make it easier to use cf-net without bootstrapping or sudo.
```

Comment:
I would drop this "in the future..." sentence, statements such as these tend to linger for a long time if we don't end up doing it. I think it belongs in a ticket instead.